src/main/scala/tilelink/Parameters.scala

```diff
@@ -1085,27 +1151,58 @@ case class TLEdgeParameters(
+  def expectsGetMasterToSlaveSafe       (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsGet(sourceId, lgSize, range)         && slave.supportsGetSafe(address, lgSize, range)
+  def expectsPutFullMasterToSlaveSafe   (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsPutPartial(sourceId, lgSize, range)  && slave.supportsPutPartialSafe(address, lgSize, range)
+  def expectsPutPartialMasterToSlaveSafe(sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsPutPartial(sourceId, lgSize, range)  && slave.supportsPutPartialSafe(address, lgSize, range)
+  def expectsHintMasterToSlaveSafe      (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsHint(sourceId, lgSize, range)        && slave.supportsHintSafe(address, lgSize, range)
+
+  def expectsAcquireTMasterToSlaveFast  (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsAcquireT(sourceId, lgSize, range)    && slave.supportsAcquireTFast(address, lgSize, range)
+  def expectsAcquireBMasterToSlaveFast  (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsAcquireB(sourceId, lgSize, range)    && slave.supportsAcquireBFast(address, lgSize, range)
+  def expectsArithmeticMasterToSlaveFast(sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsArithmetic(sourceId, lgSize, range)  && slave.supportsArithmeticFast(address, lgSize, range)
+  def expectsLogicalMasterToSlaveFast   (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsLogical(sourceId, lgSize, range)     && slave.supportsLogicalFast(address, lgSize, range)
+  def expectsGetMasterToSlaveFast       (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsGet(sourceId, lgSize, range)         && slave.supportsGetFast(address, lgSize, range)
+  def expectsPutFullMasterToSlaveFast   (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsPutFull(sourceId, lgSize, range)     && slave.supportsPutFullFast(address, lgSize, range)
+  def expectsPutPartialMasterToSlaveFast(sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsPutPartial(sourceId, lgSize, range)  && slave.supportsPutPartialFast(address, lgSize, range)
+  def expectsHintMasterToSlaveFast      (sourceId: UInt, address: UInt, lgSize: UInt, range: Option[TransferSizes] = None) = master.emitsHint(sourceId, lgSize, range)        && slave.supportsHintFast(address, lgSize, range)
+
+  //We don't use Safe vs Fast because we don't have the equivalent of address decoder for sourceIds
+// TODO (can use sourceId here to see if the specific master can support this transaction)
+  def expectsProbeSlaveToMaster      = master.supportsProbe
```

Comment:
Duality: these B channel messages from `SlaveToMaster` are:
- being routed to a client based on a `sourceId`, so they need to be passed to a helper that checks whether the _master_ that owns certain _sourceIds_ claimed to _support_ transactions of this size/type
- being sent from the manager of a particular address, so they need to be passed to a helper that checks with the _slave_ that owns certain _addresses_  claimed to _emit_ transactions of this size/type